src/nvim/drawline.c

```diff
@@ -275,6 +275,15 @@ static void draw_virt_text(win_T *wp, buf_T *buf, int col_off, int *end_col, int
     if (decor_virt_pos(item) && item->draw_col == -1) {
       bool updated = true;
       VirtTextPos pos = decor_virt_pos_kind(item);
+
+      if (pos == kVPosRightAlignTrunc) {
+        if (do_eol && (vt->width <= (right_pos - state->eol_col))) {
+          pos = kVPosRightAlign;
+        } else {
+          pos = kVPosEndOfLine;
+        }
```

Comment:
This will lead to inconsistent sorting order. `kVPosRightAlign` places virtual text from right to left, while `kVPosEndOfLine` places virtual text from left to right.

Reply:
You're absolutely right! I'll take another swipe at this

Reply:
Now I'm checking for the combined length of the eol_right_aligned virtual text, and setting the eol_col to a position that would result in combined right alignment. If the combined virtual text is too long to fit, it doesn't change the eol_col.

In either case, once it has conditionally changed the eol_col, it just treats the "eol_right_align" virtual text as "eol" virtual text.